src/components/comments/Comment/index.js

```diff
@@ -0,0 +1,128 @@
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Avatar, Grid, Typography } from '@material-ui/core';
+import moment from 'moment';
+import { useSafeIntl } from '../../../utils/useSafeIntl';
+import { MESSAGES } from './messages';
+import { useStyles } from '../styles';
+import '../../../css/index.css';
+
+const MAX_TEXT_LENGTH = 150;
+const truncateText = (text, maxLength) => {
+    if (text.length > maxLength) {
+        return `${text.substring(0, maxLength - 4)}...   `;
+    }
+    return `${text}   `;
+};
+
+const CommentText = ({
+    text,
+    hideOverflow,
+    toggle,
+    maxLength,
+    textExpand,
+    textCollapse,
+}) => {
+    const classes = useStyles();
+    const intl = useSafeIntl();
+    return (
+        <div className={classes.commentText}>
+            <p>
+                {hideOverflow && text.length > maxLength
+                    ? truncateText(text, MAX_TEXT_LENGTH)
+                    : text}{' '}
+            </p>
+            {hideOverflow && (
+                <span
+                    onClick={toggle}
+                    className={classes.toggleCommentText}
+                    role="button"
+                    tabIndex={0}
+                >
+                    {textExpand ?? intl.formatMessage(MESSAGES.textExpand)}
+                </span>
+            )}
+            {!hideOverflow && text.length > maxLength && (
+                <span
+                    onClick={toggle}
+                    className={classes.toggleCommentText}
+                    role="button"
+                    tabIndex={0}
+                >
+                    {textCollapse ?? intl.formatMessage(MESSAGES.textCollapse)}
+                </span>
+            )}
+        </div>
+    );
+};
+CommentText.propTypes = {
+    text: PropTypes.string.isRequired,
+    hideOverflow: PropTypes.bool.isRequired,
+    toggle: PropTypes.func.isRequired,
+    maxLength: PropTypes.number.isRequired,
+    textExpand: PropTypes.string,
+    textCollapse: PropTypes.string,
+};
+
+CommentText.defaultProps = {
+    textExpand: null,
+    textCollapse: null,
+};
+
+// TODO refactor style import
+// credit: https://codesandbox.io/s/comment-box-with-material-ui-10p3c?file=/src/index.js:2810-4030
+const Comment = ({ avatar, author, content, postingTime, classNames }) => {
+    const intl = useSafeIntl();
+    const defaultClasses = useStyles();
+    const [hideTextOverflow, setHideTextOverflow] = useState(
+        content.length > MAX_TEXT_LENGTH,
+    );
+    const toggleOverflowDisplay = useCallback(
+        () => setHideTextOverflow(!hideTextOverflow),
+        [hideTextOverflow],
+    );
+    const classes = classNames ?? defaultClasses;
+    return (
+        <Grid container wrap="nowrap" spacing={4}>
+            {avatar && (
+                <Grid item>
+                    {' '}
+                    <Avatar alt={author} src={avatar} />
```

Comment:
Material ui avatar is using a default icon if no src. We can also use username or last name first letter